ASCIIEngine/CollisionHandler.cs

```diff
@@ -59,17 +59,17 @@ public static void ResolveCollisions(IEnumerable<GameObject> objects)
 
         private static void ProcessRigidBody(GameObject obj, Vector2D direction)
         {
-            if (obj.Components.ContainsKey(typeof(RigidBody2D)))
+            var rBody = obj.GetComponent<RigidBody2D>();
+            if (rBody == null) 
+                return;
+            
+            if (rBody.Direction == Vector2D.Zero)
```

Comment:
What if rBody's direction is nonzero, but its velocity is zero? Why not leave only the velocity and normalize it, to get the direction?

Reply:
Resolved